job/aggregation.go

```diff
@@ -12,18 +13,25 @@ type Aggregation struct {
 
 	// Maximum numbers of concurrent download requests
 	Limit int `json:"aggr_limit"`
+
+	// Proxy url for the client to use, optional
+	Proxy string `json:"aggr_proxy"`
 }
 
 // NewAggregation creates an aggregation with the provided ID and limit.
 // If any of the prerequisites fail, an error is returned.
-func NewAggregation(id string, limit int) (*Aggregation, error) {
+func NewAggregation(id string, limit int, proxy string) (*Aggregation, error) {
 	if id == "" {
 		return nil, errors.New("Aggregation ID cannot be empty")
 	}
 	if limit <= 0 {
 		return nil, errors.New("Aggregation limit must be greater than 0")
 	}
-	return &Aggregation{ID: id, Limit: limit}, nil
+
+	if _, err := url.ParseRequestURI(proxy); proxy != "" && err != nil {
```

Comment:
We don't have to try to parse the proxy arg if it's empty, right?

Reply:
No we don't. 
I originally wanted to be 
`if (proxy != "")  && ( _, err := url.ParseRequestURI(proxy); err != nil) {`
But seems that Go does not find it reasonable enough to support this kind of expressions. 🙁

Reply:
How about something like this:

```
if proxy != "" {
  err := url.ParseRequestURI(proxy)
  if err != nil {
    return nil, errors.New("Aggregation proxy must be valid")
  }
}
```